package/samplers/smac_sampler/sampler.py

```diff
@@ -0,0 +1,274 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+
+from ConfigSpace import Categorical
+from ConfigSpace import Configuration
+from ConfigSpace import ConfigurationSpace
+from ConfigSpace import Float
+from ConfigSpace import Integer
+import numpy as np
+from optuna.distributions import BaseDistribution
+from optuna.distributions import CategoricalDistribution
+from optuna.distributions import FloatDistribution
+from optuna.distributions import IntDistribution
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+import optunahub
+from smac.acquisition.function import AbstractAcquisitionFunction
+from smac.acquisition.function import EI
+from smac.acquisition.function import LCB
+from smac.acquisition.function import PI
+from smac.facade import BlackBoxFacade
+from smac.facade import HyperparameterOptimizationFacade
+from smac.initial_design import AbstractInitialDesign
+from smac.initial_design import LatinHypercubeInitialDesign
+from smac.initial_design import RandomInitialDesign
+from smac.initial_design import SobolInitialDesign
+from smac.model.abstract_model import AbstractModel
+from smac.runhistory.dataclasses import TrialInfo
+from smac.runhistory.dataclasses import TrialValue
+from smac.runhistory.enumerations import StatusType
+from smac.scenario import Scenario
+from smac.utils.configspace import get_config_hash
+
+
+SimpleBaseSampler = optunahub.load_module("samplers/simple").SimpleBaseSampler
+
+
+def dummmy_target_func(config: Configuration, seed: int = 0) -> float:
+    # This is only a placed holder function that allows us to initialize a new SMAC facade
+    return 0
+
+
+class SMACSampler(SimpleBaseSampler):  # type: ignore
+    def __init__(
+        self,
+        search_space: dict[str, BaseDistribution],
+        n_trials: int = 100,  # This is required for initial design
```

Comment:
This value determines the number of initial value: https://github.com/automl/SMAC3/blob/main/smac/initial_design/abstract_initial_design.py#L40

In SMAC, we set the maximal number of initial design as `min(max_ration * n_trials, n_hps_per_dim * n_dim)`.
So if the value is larger than the value that we actually need, we might waste too many trials on the initial designs instead of BO iterations. On the opposite, if we set this value too small, then we might not have enough initial designs for early exploration.